drivers/display/ls0xx.c

```diff
@@ -0,0 +1,493 @@
+ *
+ * @param driver Pointer to the display driver
+ */
+static inline void toggle_vcom_bit(struct ls0xx_data *driver) {
+#if defined(CONFIG_LS0XX_EXTMODE_SERIAL)
+    extcomin_bit = extcomin_bit ? 0x00 : LS0XX_BIT_VCOM;
+#else
+    gpio_pin_toggle(driver->extcomin, LS0XX_EXTCOMIN_PIN);
+#endif
+}
+
+/**
+ * @brief Display memory contents of the display
+ *
+ * @details Display memory contents, ie actual image data 
+ *          During initialization DISP pin will be set high
+ *
+ * @param dev Pointer to the display device
+ */
+static int ls0xx_blanking_off(const struct device *dev) {
+#if defined(LS0XX_DISP_CNTRL)
+    struct ls0xx_data *driver = dev->data;
+    return gpio_pin_set(driver->disp, LS0XX_DISP_PIN, 1);
+#else
+    LOG_ERR("Unsupported");
+    return -ENOTSUP;
+#endif
+}
+
+/**
+ * @brief Display becomes completely white
+ *
+ * @details Displays complete white while but the
+ *          memory contents are retained
+ *
+ * @param dev Pointer to the display device
+ */
+static int ls0xx_blanking_on(const struct device *dev) {
+#if defined(LS0XX_DISP_CNTRL)
+    struct ls0xx_data *driver = dev->data;
+    return gpio_pin_set(driver->disp, LS0XX_DISP_PIN, 0);
+#else
+    LOG_ERR("Unsupported");
+    return -ENOTSUP;
+#endif
+}
+
+/**
+ * @brief Clear the internal memory of the display
+ *
+ * @details Clear the display memory and the frame_buf
+ *          setting it to all white
+ *
+ * @param dev Pointer to the display device
+ */
+static int ls0xx_clear(const struct device *dev) {
+
+    struct ls0xx_data *driver = dev->data;
+    int err;
+    memset(frame_buf, 0xFF, LS0XX_PANEL_HEIGHT*(((LS0XX_PANEL_WIDTH/LS0XX_PIXELS_PER_BYTE) + 2)));
+
+    uint8_t clear_cmd[2] = {LS0XX_BIT_CLEAR | LS0XX_EXTCOMIN_BIT, 0};
+    toggle_vcom_bit(driver);
+    buf[0].len = sizeof(uint16_t);
+    buf[0].buf = &clear_cmd;
+    buf_set.count = 1;
+
+    gpio_pin_set(driver->scs, LS0XX_SCS_PIN, 1);
+    err = spi_write(driver->spi_dev, &driver->spi_config, &buf_set);
+    gpio_pin_set(driver->scs, LS0XX_SCS_PIN, 0);
+
+    return err;
+}
+
+/**
+ * @brief Send required lines to the LCD
+ *
+ * @details Update the display by sending num_lines
+ *          starting from start_line.
+ *          If num_lines is 0 then only VCOM will be
+ *          toggled
+ *
+ * @param dev Pointer to the display device
+ * @param start_line Starting line number, starts with 1
+ *                   corresponds to Y co-ordinate + 1 
+ * @param num_lines Number of lines ie height to update
+ */
+static int ls0xx_update_display(const struct device *dev, 
+                                uint16_t start_line, uint16_t num_lines) {
+
+    struct ls0xx_data *driver = dev->data;
+    int err;
+    uint8_t write_cmd;
+
+    LOG_DBG("Updating lines from %d to %d", start_line, start_line+num_lines-1);
+
+    gpio_pin_set(driver->scs, LS0XX_SCS_PIN, 1);
+
+    /* Send write command along with extcomin_bit */
+    write_cmd = LS0XX_BIT_WRITECMD | LS0XX_EXTCOMIN_BIT;
```

Comment:
Add ls0xx_spi_write_cmd function.

Reply:
Do you mean a separate SPI transfer only for the write cmd? Or a separate function just to populate the buf[0] ?